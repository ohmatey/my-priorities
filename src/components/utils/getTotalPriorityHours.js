const getTotalPriorityHours = priorities => priorities.reduce((total, priority) => total + priority.hours, 0)

export default getTotalPriorityHours
