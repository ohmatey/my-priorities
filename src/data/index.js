import Immutable from 'seamless-immutable'

export const prioritiesList = Immutable([
  {
    id: 0,
    name: 'Work',
    hours: 9
  },
  {
    id: 1,
    name: 'Yoga',
    hours: 1
  },
  {
    id: 2,
    name: 'Gym',
    hours: 1
  },
  {
    id: 3,
    name: 'Family Time',
    hours: 1
  }
])
