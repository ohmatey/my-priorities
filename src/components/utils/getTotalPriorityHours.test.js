import getTotalPriorityHours from './getTotalPriorityHours'
import { prioritiesList } from '../../data/'

describe('getTotalPriorityHours function', () => {
  it('matches snapshot with empty array', () => {
    const totalHours = getTotalPriorityHours([])

    expect(totalHours).toMatchSnapshot()
  })

  it('matches snapshot with array', () => {
    const totalHours = getTotalPriorityHours(prioritiesList)

    expect(totalHours).toMatchSnapshot()
  })
})
