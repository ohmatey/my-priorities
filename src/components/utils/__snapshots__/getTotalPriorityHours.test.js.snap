// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTotalPriorityHours function matches snapshot with array 1`] = `12`;

exports[`getTotalPriorityHours function matches snapshot with empty array 1`] = `0`;
